src/Parsers/Kusto/ParserKQLStatement.cpp

```diff
@@ -95,8 +97,59 @@ bool ParserKQLTaleFunction::parseImpl(Pos & pos, ASTPtr & node, Expected & expec
             return true;
         }
     }
-    pos =  begin;
+    pos = begin;
     return false;
-};
+};*/
+
+bool ParserKQLTableFunction::parseImpl(Pos & pos, ASTPtr & node, Expected & expected)
+{
+    ParserKQLWithUnionQuery kql_p;
+    ASTPtr select;
+    ParserToken s_lparen(TokenType::OpeningRoundBracket);
+
+    auto begin = pos;
+    auto paren_count = 0;
+    String kql_statement;
+
+    if (s_lparen.ignore(pos, expected))
+    {
+        if (pos->type == TokenType::HereDoc)
+        {
+            auto kql_table_str = String(pos->begin, pos->end);
+            size_t pos1 = kql_table_str.find("$$");
+            size_t pos2 = kql_table_str.rfind("$$");
+            if (pos1 != std::string::npos && pos2 != std::string::npos && pos1 < pos2)
+                kql_statement = kql_table_str.substr(pos1+2, pos2 - pos1 -2);
+        }
+        else
+        {
+            ++paren_count;
+            auto pos_start = pos;
+            while (!pos->isEnd())
+            {
+                if (pos->type == TokenType::ClosingRoundBracket)
+                    --paren_count;
+                if (pos->type == TokenType::OpeningRoundBracket)
+                    ++paren_count;
+
+                if (paren_count == 0)
+                    break;
+                ++pos;
+            }
+            kql_statement = String(pos_start->begin, (--pos)->end);
+        }
+        ++pos;
+        Tokens token_kql(kql_statement.c_str(), kql_statement.c_str() + kql_statement.size());
+        IParser::Pos pos_kql(token_kql, pos.max_depth);
 
+        if (kql_p.parse(pos_kql, select, expected))
```

Comment:
You are still using clickhouse tokenizer - for now it maybe ok and you can leave it this way, but, again, in future it can break on kql syntax - I strongly recommend to introduce your own tokenizer (maybe even a copy of ch's) - at least on the next phase...

Reply:
yes, a dedicated  tokenizer  for KQL is implemented in next phase,  so next commit will have the KQL tokenizer